pyquil/tests/test_operator_estimation.py

```diff
@@ -599,3 +599,54 @@ def test_ratio_variance_array():
     var_b = np.array([0.05, 0.5, 5.0])
     ab_ratio_var = ratio_variance(a, var_a, b, var_b)
     np.testing.assert_allclose(ab_ratio_var, np.array([0.028125, 0.0028125, 0.00028125]))
+
+
+def test_measure_observables_noisy_asymmetric_readout_plus_one():
+    qc = get_qc('9q-qvm')
+    expt1 = ExperimentSetting(TensorProductState(plusX(0)), sX(0))
+    expt2 = ExperimentSetting(TensorProductState(plusY(0)), sY(0))
+    expt3 = ExperimentSetting(TensorProductState(plusZ(0)), sZ(0))
+    p = Program()
+    p.define_noisy_readout(0, p00=0.99, p11=0.80)
+    qubs = [0]
+    tomo_expt = TomographyExperiment(settings=[expt1, expt2, expt3], program=p, qubits=qubs)
+
+    for res in measure_observables(qc, tomo_expt, n_shots=1000,
+                                   readout_symmetrize='exhaustive',
+                                   calibrate_readout='plus-eig'):
+        assert np.isclose(1.0, res.expectation, atol=1.e-1)
+
+
+def test_measure_observables_noisy_asymmetric_readout_minus_one():
```

Comment:
I don't think this need to be a different test. Let's put all these 1Q tests on the same test.

A more meaningful test to implement would be 2Q correlations, or maybe larger correlations as well -- these are important use cases that need to be capture.

Reply:
On that topic, I would also run tests where the expectation should be zero, e.g.,
```
expt1 = ExperimentSetting(TensorProductState(plusX(0)), sZ(0))
```
and making sure it behaves as expected, both with symmetrization/calibration and without.

Reply:
> @msohaibalam I think there is a bug. If you have two qubits, and they have different error rates, but you only measure one of them, the raw expectation and the calibration seem off.
> 
> ```python
> expt5 = ExperimentSetting(TensorProductState(plusZ(qubs[0])*plusZ(qubs[1])), sZ(qubs[0]))
> p = Program()
> p.define_noisy_readout(0, 0.999, 0.85)
> p.define_noisy_readout(1, 0.999, 0.75)
> tomo_experiment5 = TomographyExperiment(settings=[expt4]*runs, program=p, qubits=[qubs[0],qubs[1]])
> 
> raw_e = np.zeros(runs)
> obs_e = np.zeros(runs)
> cal_e = np.zeros(runs)
> 
> for idx,res in enumerate(measure_observables(qc, 
>                                          tomo_experiment5, 
>                                          n_shots=10000, 
>                                          active_reset=True,
>                                          readout_symmetrize='exhaustive', 
>                                          calibrate_readout='plus-eig')):
>     raw_e[idx] = res.raw_expectation
>     obs_e[idx] = res.expectation
>     cal_e[idx] = res.calibration_expectation
> 
> np.mean(raw_e), np.mean(obs_e), np.mean(cal_e) 
> ```
> 
> which yields something close to `(0.63,1.0,0.63)`, when it should yield something close to `(0.75,1.0,0.75)`
> 
> TBH I can't tell if it is a QVM bug or a `measure_observables` bug

@marcusps I don't see this bug. I think you might want to replace `expt4` with `expt5` in the definition of `tomo_experiment5`. But also, I don't agree that it should yield something close to `(0.75,1.0,0.75)`. I think it should instead be close to `(0.849, 1.0, 0.849)`, which is what I've added a test for.

Reply:
^ Those expectations should be zero with no symmetrization/calibration only if there's also no readout errors. In the case of readout errors, their non-zero values are predictable in terms of the readout errors though. I've added tests for this and related scenarios.